proxy/src/main/java/com/velocitypowered/proxy/connection/backend/TransitionSessionHandler.java

```diff
@@ -137,12 +137,12 @@ public boolean handle(JoinGamePacket packet) {
           smc.setActiveSessionHandler(StateRegistry.PLAY,
               new BackendPlaySessionHandler(server, serverConn));
 
-          // Clean up disabling auto-read while the connected event was being processed.
-          smc.setAutoReading(true);
-
           // Now set the connected server.
           serverConn.getPlayer().setConnectedServer(serverConn);
 
+          // Clean up disabling auto-read while the connected event was being processed.
+          smc.setAutoReading(true);
```

Comment:
Would you mind expanding the comment above this line to include your description (or similar)?